### PR TITLE
[HLS-3256] - propagate context from traceparent header

### PR DESCRIPTION
**Pull request by LironKS**

### CODE REVIEW

File: proxy-libs/helioslambda/lambda.go [hunk 1]
```diff
@@ -2,21 +2,44 @@ package helioslambda
...
+func heliosEventToCarrier (eventJSON []byte) propagation.TextMapCarrier {
+	var headers HttpHeaders
+	err := json.Unmarshal(eventJSON, &headers)
+	if err != nil {
+		return propagation.HeaderCarrier{"": []string{""}}
+	} else {
+			if val, ok := headers.Headers["Traceparent"]; ok  {
+				return propagation.HeaderCarrier{"Traceparent": []string{val}}
+			} else if val, ok = headers.Headers["traceparent"]; ok {
+				return propagation.HeaderCarrier{"traceparent": []string{val}}
```
Comment (nozik) [inline proxy-libs/helioslambda/lambda.go:28]: this should be uppercase

File: proxy-libs/helioslambda/lambda.go [hunk 2]
```diff
@@ -2,21 +2,44 @@ package helioslambda
...
 var InstrumentedSymbols = [...]string{"Start", "StartWithContext", "StartWithOptions"}
 
+type HttpHeaders struct {
+	Headers map[string]string `json:"headers"`
+}
+func heliosEventToCarrier (eventJSON []byte) propagation.TextMapCarrier {
+	var headers HttpHeaders
+	err := json.Unmarshal(eventJSON, &headers)
+	if err != nil {
+		return propagation.HeaderCarrier{"": []string{""}}
```
Comment (nozik) [inline proxy-libs/helioslambda/lambda.go:23]: you can remove this err != nil, the return in the last statement takes care of that

**Review by nozik — CHANGES_REQUESTED** [review]

File: proxy-libs/helioslambda/lambda.go (hunk 2 above)
Comment (LironKS) [reply proxy-libs/helioslambda/lambda.go:23]: but I'm not sure it will not throw me if I'll try to acsess headers

Comment (nozik) [reply proxy-libs/helioslambda/lambda.go:23]: What I'm saying is you can just check if err == nil, else return propagation.HeaderCarrier{"": []string{""}}

**Review by nozik — APPROVED** [review]

**Pull request merged**